### PR TITLE
Add Typescript definitions

### PR DESCRIPTION
**Pull request by ahanriat**
Hi Guys 👋

Just a quick PR to add **Typescript** definitions.
Hope you'll like it !

Cheers

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import { ViewProperties } from 'react-native';
+
+export interface BlurViewProperties {
+  blurType: 'xlight' | 'light' | 'dark'
+    // tvOS only
+    | 'extraDark' | 'regular' | 'prominent';
+  blurAmount?: number; // 0 - 100
```
Comment (Kureev) [inline index.d.ts]: Can we be more precise about the range?

**Comment by Kureev** [comment]
Hi @ahanriat!

Nice! Thank you for the PR! I have one question regarding the range of the `number`. We know in advance that the range is from 1 to 100, so maybe we can enforce it on the type level?

File: index.d.ts (hunk 1 above)
Comment (ahanriat) [inline index.d.ts]: Hi @Kureev, the only possible way to do it currently would be to have a enum like:
```
blurAmount?: 0 | 1 | 2 | 3 | ... | 99 | 100;
```
This issue on Typescript is exactly what we are looking for, but it looks like there is currently no clean way to do it, appart using an enum https://github.com/Microsoft/TypeScript/issues/15480 👀

Comment (Kureev) [inline index.d.ts]: Ah, got it! Thanks for the insight! Then I think it isn't a problem

**Comment by Kureev** [comment]
Thank you for this awesome contribution 🙌

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import { ViewStyle } from 'react-native';
+
+export interface BlurViewProperties {
+  blurType: 'xlight' | 'light' | 'dark'
+    // tvOS only
+    | 'extraDark' | 'regular' | 'prominent';
+  blurAmount?: number; // 0 - 100
```
Comment (arnibarnason) [inline index.d.ts]: Shouldn't viewRef be in there as well?

Comment (ahanriat) [inline index.d.ts]: @arnibarnason doesn't the `extends React.Component` does this?

**Pull request merged**